bucket-version-mismatch/README.md

```diff
@@ -0,0 +1,109 @@
+coll.validate();
+// The warnings field detects mixed-schema buckets. 
+{
+"ns" : "db.system.buckets.coll",
+...
+"errors" : [
+    Detected one or more documents in this collection incompatible with time-series
+    specifications. For more info, see logs with log id 6698300.,
+    ...
+],
+...
+}
+```
+with the logs:
+```
+..."c":"STORAGE",  "id":6698300, "ctx":"conn9","msg":"Document is not compliant with time-series specifications","attr":{"namespace":"[...]","recordId":[...],"reason":{"code":2,"codeName":"BadValue","errmsg":"Time-series bucket 't' field is not in ascending order"}}...
+```
+or 
+```
+..."c":"STORAGE",  "id":6698300, "ctx":"conn9","msg":"Document is not compliant with time-series specifications","attr":{"namespace":"[...]","recordId":[...],"reason":{"code":2,"codeName":"BadValue","errmsg":"Time-series bucket is v3 but has its measurements in-order on time"}}...
+```
+Example `validate` run on a sharded cluster:
+```
+// Call validate on mongos for sharded clusters.
+coll.validate();
+// The warnings field detects mixed-schema buckets.
+// For sharded clusters, this output is an object with a result for every shard in 
+// the "raw" field.
+{
+	"ns" : "db.system.buckets.coll",
+	...
+"errors" : [
+    Detected one or more documents in this collection incompatible with time-series
+    specifications. For more info, see logs with log id 6698300.,
+    ...
+],
+...
+"raw" : {
+	"shard-0-name" : {
+		"ns" : "db.system.buckets.coll"
+		...
+"errors" : [
+    Detected one or more documents in this collection incompatible with time-series
+    specifications. For more info, see logs with log id 6698300.,
+    ...
+],
+...
+},
+"shard-1-name" : { ... }, ...
+}
+}
+```
+with the logs:
+```
+..."c":"STORAGE",  "id":6698300, "ctx":"conn9","msg":"Document is not compliant with time-series specifications","attr":{"namespace":"[...]","recordId":[...],"reason":{"code":2,"codeName":"BadValue","errmsg":"Time-series bucket 't' field is not in ascending order"}}...
+```
+or 
+```
+..."c":"STORAGE",  "id":6698300, "ctx":"conn9","msg":"Document is not compliant with time-series specifications","attr":{"namespace":"[...]","recordId":[...],"reason":{"code":2,"codeName":"BadValue","errmsg":"Time-series bucket is v3 but has its measurements in-order on time"}}...
+```
+# Remediation
+## Rewrite Bucket Version in a Time Series Collections
+While the script is running, the performance of operations on the time-series collection may be impacted. The script does a scan of the whole collection and performs updates on impacted buckets, which may result in a large load if many buckets are affected.  
+At a high level, the script remediates performance by updating the bucket's version from v2 to v3 and vice versa if a v2 bucket has unsorted data, and if a  from the mixed-schema format to the older schema.  The rewrite is done by unpacking the measurements of the problematic mixed-schema buckets and inserting those measurements back into the collection.
```

Comment:
"The script remediates performance" 

Should this say, "the script remediates validation issues by...."?

Reply:
nit: there is a double space here: "and if a  from" between 'a' and 'from'